### ISSUE TITLE
[svg-2019] must test as you commit is too strict

### ISSUE THREAD

**Issue by frivoal** [w3c/charter-drafts#186]
[The svg 2019 charter](https://w3c.github.io/charter-drafts/svg-2019.html) requires a "test as you commit policy", which is defined as (see also https://github.com/w3c/charter-drafts/issues/185):

> All normative spec changes are generally expected to have a corresponding pull request in web-platform-tests, either in the form of new tests or modifications to existing tests, or must include the rationale for why test updates are not required for the proposed update.

If I'm reading this correctly, if you cannot justify tests being unneeded, then it is forbidden to merge the spec PR. I believe this means that if a PR should have tests, and is lacking them because nobody is signing up to write the tests, the WG chair would be forbidden by the charter to decide to merge anyway.

While I agree having tests is important, that seems too strict to me. The SVG WG tends to be understaffed, and this probably means that one cannot (effectively) be an editor without also signing up to write the tests. This seems discouraging.

The same rule, downgraded to a should, would be much better.

Alternatively, it is already meant as a should, but in that case, I'd suggest rephrasing. The sentence construct "should/may do A, or must do B" seems to make B the only allowed exception to A.

**Comment by AmeliaBR** [comment]
I don't read this as a "must have tests before merging". I read it as "if tests won't be required, that must be noted on the PR", otherwise tests are expected.

The "must" is nonetheless still a little strong. As Florian notes, we tend to be understaffed at the SVG WG and although we have been reviewing all normative edits since we went to CR, sometimes people merge and then remember about other things after.

**Comment by svgeesus** [comment]
Note that this wording [also comes from the W3C charter template](https://github.com/w3c/charter-drafts/blob/gh-pages/charter-template.html#L164) and is seen as good practice for all W3C WGs moving forward.

**Comment by frivoal** [comment]
I think testing as you go is good practice. I think denying the chairs the ability to accept a change in a spec unless there are tests or tests are not needed at all is misguided. I'm not 100% sure that the language intends to be that strong, but I can read it that way, and I think that's a mistake.

The wording you linked to in the charter template is different, and much more reasonable, as it uses a simple should. But then it goes on to link to the external document that uses a must, and that's unfortunate.

**Comment by svgeesus** [comment]
The [external document linked to](https://www.w3.org/2019/02/testing-policy.html) has now changed and, in particular, allows a `needs tests` tag for a PR that needs tests. That seems to give the chairs and editors adequate flexibility (and is current practice in CSS WG (with a [slightly different tagname](https://github.com/w3c/csswg-drafts/labels/Needs%20Testcase%20%28WPT%29), `Needs Testcase (WPT)`).

**Comment by frivoal** [comment]
(commenting in the closed issue, as I don't seem to have permission to reopen)

I don't consider this issue addressed:

* Just because we have documented a way (using labels) of marking which spec PRs don't have tests yet does not mean we have added the permission to merge such spec PRs, especially when the previous paragraph specifically defines lacking tests as being a reason not to merge PRs.

* The CSS-WG is not a useful reference here: it does not follow a mandatory test as you commit approach, neither in charter nor in practice. It documents what tests are needed as you commit, and writes them eventually. I think is a good practice, and would recommend it for the SVG-WG, and it is precisely because the proposed charter would disallow that that I don't like it.

* The first paragraph boils down to "should do A, or must do B", which I read as B being the only allowed exception to A. From there, my reading is that if a spec PR is possible and desirable to test, then test MUST be written, and the spec PR cannot be merged until they are. This is the policy that some groups follow, so I don't believe that this interpretation can be rejected as unreasonable. I do think it would be a bad rule to have for the SVG WG, since it is quite severely understaffed, and applying this rule strictly is likely to increase the number of pending PRs that cannot be merged (despite everybody agreeing that they should), without appreciably increasing the number of test.

* While the WG has been out of charter since end of January, it has continued to operate. Since then, some spec PRs have been merged without corresponding tests PRs being merged, nor with statements of why tests were not required. Will these changes be rolled back when the new charter kicks in (at least until the corresponding tests have been written)? I hope not, but why have rules if they're not being followed? Also, the working mode follow by the group right now (document what needs to be tested as you make changes to the spec, write the tests eventually when someone has time) is how I expect the group to operate most of the time given the lack of test writers. If we're trying to ban that, I think we're wrong, and if we're not, I think the charter is broken.
